sktime/forecasting/arch.py

```diff
@@ -0,0 +1,307 @@
+
+        self.mean = mean
+        self.lags = lags
+        self.vol = vol
+        self.p = p
+        self.o = o
+        self.q = q
+        self.power = power
+        self.dist = dist
+        self.hold_back = hold_back
+        self.rescale = rescale
+    
+        super(ARCH, self).__init__()
+
+
+    def _fit(self, y, X=None, **fit_params):
+        """Fit the training data to the estimator
+
+        Parameters
+        ----------
+        y : guaranteed to be of a type in self.get_tag("y_inner_mtype")
+            Time series to which to fit the forecaster.
+            if self.get_tag("scitype:y")=="univariate":
+                guaranteed to have a single column/variable
+            if self.get_tag("scitype:y")=="multivariate":
+                guaranteed to have 2 or more columns
+            if self.get_tag("scitype:y")=="both": no restrictions apply
+        X : optional (default=None)
+            guaranteed to be of a type in self.get_tag("X_inner_mtype")
+            Exogeneous time series to fit to.
+
+        update_freq : int, optional
+            Frequency of iteration updates.  Output is generated every
+            update_freq iterations. Set to 0 to disable iterative output.
+        disp : {bool, "off", "final"}
+            Either 'final' to print optimization result or 'off' to display
+            nothing. If using a boolean, False is "off" and True is "final"
+        starting_values : np.ndarray, optional
+            Array of starting values to use.  If not provided, starting values
+            are constructed by the model components.
+        cov_type : str, optional
+            Estimation method of parameter covariance.  Supported options are
+            'robust', which does not assume the Information Matrix Equality
+            holds and 'classic' which does.  In the ARCH literature, 'robust'
+            corresponds to Bollerslev-Wooldridge covariance estimator.
+        show_warning : bool, optional
+            Flag indicating whether convergence warnings should be shown
+        first_obs : {int, str, datetime, Timestamp}
+            First observation to use when estimating model
+        last_obs : {int, str, datetime, Timestamp}
+            Last observation to use when estimating model
+        tol : float, optional
+            Tolerance for termination.
+        options : dict, optional
+            Options to pass to scipy.optimize.minimize.  Valid entries
+            include 'ftol', 'eps', 'disp', and 'maxiter'.
+        backcast : {float, np.ndarray}, optional
+            Value to use as backcast. Should be measure \sigma^2_0
+            since model-specific non-linear transformations are applied to
+            value before computing the variance recursions.
+
+        Returns
+        -------
+        self : returns an instance of self
+        """
+        from arch import arch_model as _ARCH
+
+        self.y = y
```

Comment:
you don´t need to write y, X to self. You don´t use it later, do you?

Reply:
Actually no, will change this.

Reply:
Actually no, I will change that.